File: Fotoplastykon.Tools/Pager/Pager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Fotoplastykon.Tools.Pager
{
    public class Pager : IPager
    {
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public int TotalRows { get; set; }
    }
}

File: fotoplastykon-web-admin/src/interfaces/pager.ts
export class Pager
{
    constructor(index: number, size: number) {
        this.pageIndex = index;
        this.pageSize = size;
    }

    public pageIndex : number;
    public pageSize : number;
    public totalRows : number = 1;

    public get totalPages(): number {
        return Math.ceil(this.totalRows / this.pageSize);
    }

    public setTotalRows(rowsCount: number) {
        this.totalRows = rowsCount;
    }
    public setPageIndex(index: number) {
        this.pageIndex = index;
    }
    public setPageSize(size: number) {
        this.pageSize = size;
    }
}

export interface PaginationResult<T>
{
    pager: Pager;
    items: T[];
}

